taqueria-vscode-extension-web-ui/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { AllEditors } from './AllEditors';
+import { MichelineEditor } from './data-editors/MichelineEditor';
+import { InteropMessageInterface } from './interopTypes';
+
+const state = {
+	interop: undefined as undefined | InteropMessageInterface,
+	onChangeCallback: (x: InteropMessageInterface) => {/* empty */},
+	logs: [] as { message: string; data?: unknown }[],
+};
+
+// const log = (message: string, data?: unknown) => {
+// 	state.logs.push({ message, data });
+// };
+// console.log = log;
+
+const setGlobalInteropMessageInterface = (x: InteropMessageInterface) => {
+	state.interop = x;
+	state.onChangeCallback(x);
+};
+const w = window as unknown as Record<string, unknown>;
+w.setGlobalInteropMessageInterface = setGlobalInteropMessageInterface;
+
+export const App = () => {
+	const [deps, setDeps] = useState(state.interop);
+	useEffect(() => {
+		state.onChangeCallback = x => {
+			setDeps(x);
+		};
+		return () => {
+			state.onChangeCallback = () => {};
+		};
+	}, []);
+
+	console.log('App RENDER', { deps, interop: state.interop });
+
+	return (
+		<>
+			<EditorSelector deps={deps} />
+			<DebugView />
+		</>
+	);
+};
+
+const EditorSelector = ({ deps }: { deps: undefined | InteropMessageInterface }) => {
+	if (deps?.view === 'AllEditors') {
+		return <AllEditors />;
+	}
+	if (deps?.view === 'MichelineEditor') {
+		return <MichelineEditor {...deps} />;
+	}
+
+	return <DefaultView />;
+};
+
+const DebugView = () => {
+	const DEBUG = true;
+	if (!DEBUG) {
```

Comment:
This condition will always evaluate to false. Is this temporary code?

Reply:
I assume this to explicitly enable/disable debugging in code, but @ricklove might know better.

Reply:
It would be better to use `.env` variable for enabling debug mode or debug statements. Easier than changing const in code 😉